cleancode/src/main/java/com/epam/cleancode/comments/MortgageInstallmentCalculator.java

```diff
@@ -2,38 +2,32 @@
 
 public class MortgageInstallmentCalculator {
 
-    /**
-     *
-     * @param p principal amount
-     * @param t term of mortgage in years
-     * @param r rate of interest
-     * @return monthly payment amount
-     */
-    public static double calculateMonthlyPayment(
-            int p, int t, double r) {
+    private static final String MESSAGE_BAD_INPUT_PARAMETERS = "Negative values are not allowed";
+    private static final int AMOUNT_MONTHS_IN_YEAR = 12;
 
-        //cannot have negative loanAmount, term duration and rate of interest
-        if (p < 0 || t <= 0 || r < 0) {
-            throw new InvalidInputException("Negative values are not allowed");
+    public static double calculateMonthlyPayment(int startAmount, int termInYears, double percent) {
+        inputValidation(startAmount, termInYears, percent);
+        if (haveValidParameters(startAmount, termInYears, percent)) {
```

Comment:
I want to resolve your homework as soon as possible, but do you check your code before committing? 

I don't see function `haveValidParameters` in this class.